test/drawers/drawerTests.ts

```diff
@@ -1,137 +1,206 @@
-      let steps = [ds1, ds2, ds3];
-      drawer.draw([], steps);
-      assert.deepEqual(timings, [0, 20, 30], "setTimeout called with appropriate times");
-    });
-
-    it("selectionForIndex()", () => {
-      let svg = TestMethods.generateSVG(300, 300);
-      let drawer = createMockDrawer(null);
-      drawer.renderArea(svg.append("g"));
-      drawer.selector = () => "circle";
-      let data = [{one: 2, two: 1}, {one: 33, two: 21}, {one: 11, two: 10}];
-      let circles = drawer.renderArea().selectAll("circle").data(data);
-      circles.enter().append("circle").attr("cx", (datum: any) => datum.one).attr("cy", (datum: any) => datum.two).attr("r", 10);
-      let selection = drawer.selectionForIndex(1);
-      assert.strictEqual(selection.node(), circles[0][1], "correct selection gotten");
-      svg.remove();
-    });
-
-    it("totalDrawTime()", () => {
-      let svg = TestMethods.generateSVG(300, 300);
-      let drawer = createMockDrawer(null);
-
-      let dataObjects = 9;
-      let stepDuration = 987;
-      let stepDelay = 133;
-      let startDelay = 245;
-
-      let expectedAnimationDuration = startDelay + (dataObjects - 1) * stepDelay + stepDuration;
-
-      let data = Plottable.Utils.Array.createFilledArray({}, dataObjects);
-
-      let attrToProjector: Plottable.AttributeToProjector = null;
-
-      let animator = new Plottable.Animators.Easing();
-      animator.maxTotalDuration(Infinity);
-      animator.stepDuration(stepDuration);
-      animator.stepDelay(stepDelay);
-      animator.startDelay(startDelay);
-
-      let mockDrawStep = [{attrToProjector: attrToProjector, animator: animator}];
-
-      let drawTime = drawer.totalDrawTime(data, mockDrawStep);
-
-      assert.strictEqual(drawTime, expectedAnimationDuration, "Total Draw time");
-
-      svg.remove();
-
+    describe("animation timings", () => {
+      class MockAnimator implements Plottable.Animator {
+        private _time: number;
+        private _callback: Function;
+
+        constructor(time: number, callback?: Function) {
+          this._time = time;
+          this._callback = callback;
+        }
+
+        public totalTime(numberOfIterations: number) {
+          return this._time;
+        }
+
+        public animate(selection: any, attrToProjector: Plottable.AttributeToProjector): any {
+          if (this._callback) {
+            this._callback();
+          }
+          return selection;
+        }
+      }
+
+      let oldTimeout: any;
+      let timings: number[] = [];
+      let svg: d3.Selection<void>;
+
+      before(() => {
+        oldTimeout = Plottable.Utils.Window.setTimeout;
+        Plottable.Utils.Window.setTimeout = function(f: Function, time: number, ...args: any[]) {
+          timings.push(time);
+          return oldTimeout(f, time, args);
+        };
+      });
+
+      after(() => {
+        Plottable.Utils.Window.setTimeout = oldTimeout;
+      });
+
+      beforeEach(() => {
+        timings = [];
```

Comment:
You're right. It probably shouldn't be initialized there, since it needs to get reset before each run.